docs/patching/patch_attribute/index.rst

```diff
@@ -41,3 +41,8 @@ Properties are tricky to patch because of the quirky mechanics that `Python's De
       with_property = WithProperty()
       self.patch_attribute(with_property, "prop", "mock")
       self.assertEqual(with_property.prop, "mock")
+
+Type Validation
+---------------
+
+patch_attribute implements type validation. When you use it, the mock will raise ``TypeCheckerError`` if it is called with a type that does not match the original attribute type. This feature can be disabled with ``type_validation=False``
```

Comment:
I think this phrase would be fit for mock_callable: patch_attribute works exclusively for non-callable attributes.

Suggesstion: `patch_attribute` implements type validation and will only allow attributes to be mocked with values of types that match the available typing annotation. `TypeCheckerError` will be raised if a bad type is given. This check is expected to work 100% of times, but if it does not (possibly a  bug in TestSlide, please report!), you can disable it by doing `patch_attribute(obj, "attr", type_validation=False)`.